File: Data/Python/Encapsulation/Encapsulation.py
'''
#### Name:  Encapsulation
Link: [link]()

__ double underscore
'''

class Shift:
    def __init__(self):
        self.__manager = ''
        self.size = 0
    
    def setter(self,manager_name, size):
        self.__manager = manager_name
        self.size  = size
    
    def getter(self):
        return self.__manager + " with size "+ str(self.size)


s = Shift()

s.setter('nishan', 5)
print(s.getter())


print(s.size)
# Can't access
# print(s.__manager)

File: Data/Java/Encapsulation/Encapsulation.java
/*
#### Name:  Encapsulation
Link: [link]()

1) Hiding members of a class
3) That couldn't be changed or interferred with
2) That can be accessed using objects, using getter and setter methods

*/

class Staff{
    private String incharge;
    String  class_name  = "staff";

    void setter(String name){
        incharge = name;
    }

    String getter(){
        return incharge;
    }

}

class Encapsulation{
    public static void main(String[] args){
        Staff s = new Staff();
        s.setter("Nishan");

        // System.out.println(s.incharge);  // Staff.incharge is not visible

        System.out.println(s.getter());

    }
}